File: Swift/백준/Bronze/10757. 큰 수 A＋B/큰 수 A＋B.swift
import Foundation

// 두 개의 문자열을 자릿수별로 더하는 함수
func addLargeNumbers(_ A: String, _ B: String) -> String {
    let arrA = Array(A.reversed()) // A를 뒤집어서 배열로 변환
    let arrB = Array(B.reversed()) // B를 뒤집어서 배열로 변환

    var result = ""
    var carry = 0
    let maxLength = max(arrA.count, arrB.count)

    for i in 0..<maxLength {
        let digitA = i < arrA.count ? Int(String(arrA[i]))! : 0
        let digitB = i < arrB.count ? Int(String(arrB[i]))! : 0
        let sum = digitA + digitB + carry
        carry = sum / 10
        result.append(String(sum % 10))
    }

    if carry > 0 {
        result.append(String(carry))
    }

    return String(result.reversed()) // 결과를 뒤집어서 반환
}

// 입력받기
if let input = readLine() {
    let numbers = input.split(separator: " ").map { String($0) }
    let A = numbers[0]
    let B = numbers[1]
    
    // 결과 출력
    print(addLargeNumbers(A, B))
}

File: Rust/백준/Bronze/10757. 큰 수 A＋B/큰 수 A＋B.rs
use std::io::{self, Write};

fn add_large_numbers(a: &str, b: &str) -> String {
    let mut carry = 0;
    let mut result = String::new();

    let a_digits = a.chars().rev();
    let b_digits = b.chars().rev();

    let mut a_iter = a_digits.peekable();
    let mut b_iter = b_digits.peekable();

    while a_iter.peek().is_some() || b_iter.peek().is_some() || carry != 0 {
        let a_digit = a_iter.next().and_then(|ch| ch.to_digit(10)).unwrap_or(0);
        let b_digit = b_iter.next().and_then(|ch| ch.to_digit(10)).unwrap_or(0);

        let sum = a_digit + b_digit + carry;
        carry = sum / 10;
        result.push(std::char::from_digit(sum % 10, 10).unwrap());
    }

    result.chars().rev().collect()
}

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();

    let numbers: Vec<&str> = input.trim().split_whitespace().collect();
    let a = numbers[0];
    let b = numbers[1];

    let result = add_large_numbers(a, b);
    println!("{}", result);
}